frontend/src/components/stats/StatsComponent.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+    Box,
+    Grid,
+    LoadingOverlay,
+    useComputedColorScheme,
+} from "@mantine/core";
+import { useState } from "react";
+import { useDisclosure } from "@mantine/hooks";
+import classes from "./StatsComponent.module.css";
+import GraphExample from "./GraphExample";
+import clsx from "clsx";
+
+export type StatRenderType = "box" | "line";
+export type StatRenderBundleProps = {
+    queryPath: string;
+    renderType: StatRenderType;
+};
+
+// todo - query the backend /stats/{queryPath} and get back some data
+// todo - use the data above to update an internal state which re-renders the component
+// todo - display some placeholder while the above takes place
```

Comment:
i think i wrote these todos, feel free to delete them if they're no longer relevant. I don't need them myself anymore